vault/diagnose/tls_verification_test.go

```diff
@@ -276,6 +276,27 @@ func TestTLSNoRoot(t *testing.T) {
 	}
 }
 
+func TestTLSSelfSignedCert(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./test-fixtures/selfSignedCert.pem",
+				TLSKeyFile:                    "./test-fixtures/selfSignedCertKey.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	_, errs := ListenerChecks(context.Background(), listeners)
+	if errs != nil {
```

Comment:
Could you do an explicit check here to make sure the right error is returned for this test?